modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/data/sample/SampleDataProviderMediator.java

```diff
@@ -96,6 +98,28 @@ public Message getSampleData(ParameterValueResolver parameterValueResolver,
                                Supplier<Object> connectionSupplier,
                                Supplier<Object> configurationSupplier)
       throws SampleDataException {
+    return getSampleData(parameterValueResolver, connectionSupplier, configurationSupplier, null);
+  }
+
+  /**
+   * Resolves the sample data
+   *
+   * @param parameterValueResolver  parameter resolver required if the associated {@link SampleDataProvider} requires
+   *                                the value of parameters from the same parameter container.
+   * @param connectionSupplier      supplier of connection instances related to the container and used, if necessary,
+   *                                by the {@link SampleDataProvider}
+   * @param configurationSupplier   supplier of connection instances related to the container and used, if necessary,
+   *                                by the  {@link SampleDataProvider}
+   * @param connectionProvider      the connection provider in charge of providing the connection given by the connection supplier.
+   *
+   * @return a {@link Message} carrying the sample data
+   * @throws SampleDataException if an error occurs resolving the sample data
+   */
+  public Message getSampleData(ParameterValueResolver parameterValueResolver,
+                               Supplier<Object> connectionSupplier,
+                               Supplier<Object> configurationSupplier,
+                               ConnectionProvider connectionProvider)
```

Comment:
you're already providing the config. You should expose this in the API

Reply:
I did not understand this comment, could you rephrase?

Reply:
In order to give me the connection provider, the caller needed to create the configuration. What I can do is to receive a Supplier<ConnectionProvider> and only do it only if necessary.

The util method that makes the refresh of the token will also need to receive the Supplier instead of the provider directly.